File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.ts
function totalNumbers(digits: number[]): number {
    const freq = new Array(10).fill(0);
    for (const d of digits) freq[d]++;
    let count = 0;

    for (let h = 1; h <= 9; h++) {
        if (freq[h] === 0) continue;
        freq[h]--;
        for (let t = 0; t <= 9; t++) {
            if (freq[t] === 0) continue;
            freq[t]--;
            for (let u = 0; u <= 8; u += 2) {
                if (freq[u] > 0) count++;
            }
            freq[t]++;
        }
        freq[h]++;
    }
    return count;
}

File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.rs
impl Solution {
    pub fn total_numbers(digits: Vec<i32>) -> i32 {
        let mut freq = [0; 10];
        for d in digits {
            freq[d as usize] += 1;
        }
        let mut count = 0;

        for h in 1..=9 {
            if freq[h] == 0 { continue; }
            freq[h] -= 1;

            for t in 0..=9 {
                if freq[t] == 0 { continue; }
                freq[t] -= 1;
                
                for u in (0..=8).step_by(2) {
                    if freq[u] > 0 { count += 1; }
                }
                freq[t] += 1;
            }
            freq[h] += 1;
        }
        count
    }
}